Unify copies of SDKVariant.xcconfig

ddkilzer opened the pull request:
#### 024ecd89c7692cbcdcdbec9275e06c498ca73dcb
<pre>
Unify copies of SDKVariant.xcconfig
<a href="https://bugs.webkit.org/show_bug.cgi?id=244374">https://bugs.webkit.org/show_bug.cgi?id=244374</a>
&lt;rdar://99169843&gt;

Reviewed by Alexey Proskuryakov.

- Create Configurations folder and move a copy of
  SDKVariant.xcconfig into it.
- Add Configurations folder to WebKit.xcworkspace.
- Update `installsrc` target in Makefile to copy Configurations
  folder.
- Remove duplicate copies of SDKVariant.xcconfig in each
  project.
- Remove SDKVariant.xcconfig from Xcode projects.
- Change references to SDKVariant.xcconfig to use a relative
  path to the single copy.
- Remove SDKVariantChecker from check-webkit-style since it&apos;s no
  longer needed.

* Configurations/SDKVariant.xcconfig: Rename from Source/JavaScriptCore/Configurations/SDKVariant.xcconfig.
* Makefile:
(installsrc):
* Source/JavaScriptCore/Configurations/Base.xcconfig:
* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/ThirdParty/ANGLE/ANGLE.xcodeproj/project.pbxproj:
* Source/ThirdParty/ANGLE/Configurations/Base.xcconfig:
* Source/ThirdParty/ANGLE/Configurations/SDKVariant.xcconfig: Remove.
* Source/ThirdParty/gtest/xcode/Config/General.xcconfig:
* Source/ThirdParty/gtest/xcode/Config/SDKVariant.xcconfig: Remove.
* Source/ThirdParty/libwebrtc/Configurations/Base.xcconfig:
* Source/ThirdParty/libwebrtc/Configurations/SDKVariant.xcconfig: Remove.
* Source/ThirdParty/libwebrtc/libwebrtc.xcodeproj/project.pbxproj:
* Source/WTF/Configurations/Base.xcconfig:
* Source/WTF/Configurations/SDKVariant.xcconfig: Remove.
* Source/WTF/WTF.xcodeproj/project.pbxproj:
* Source/WebCore/Configurations/Base.xcconfig:
* Source/WebCore/Configurations/SDKVariant.xcconfig: Remove.
* Source/WebCore/PAL/Configurations/Base.xcconfig:
* Source/WebCore/PAL/Configurations/SDKVariant.xcconfig: Remove.
* Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebGPU/Configurations/Base.xcconfig:
* Source/WebGPU/Configurations/SDKVariant.xcconfig: Remove.
* Source/WebGPU/WebGPU.xcodeproj/project.pbxproj:
* Source/WebInspectorUI/Configurations/Base.xcconfig:
* Source/WebInspectorUI/Configurations/SDKVariant.xcconfig: Remove.
* Source/WebInspectorUI/WebInspectorUI.xcodeproj/project.pbxproj:
* Source/WebKit/Configurations/Base.xcconfig:
* Source/WebKit/Configurations/SDKVariant.xcconfig: Remove.
* Source/WebKit/SwiftOverlay/Configurations/Base.xcconfig:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/mac/Configurations/Base.xcconfig:
* Source/WebKitLegacy/mac/Configurations/SDKVariant.xcconfig: Remove.
* Source/bmalloc/Configurations/Base.xcconfig:
* Source/bmalloc/Configurations/SDKVariant.xcconfig: Remove.
* Source/bmalloc/bmalloc.xcodeproj/project.pbxproj:
* Tools/ContentExtensionTester/Configurations/Base.xcconfig:
* Tools/ContentExtensionTester/Configurations/SDKVariant.xcconfig: Remove.
* Tools/DumpRenderTree/DumpRenderTree.xcodeproj/project.pbxproj:
* Tools/DumpRenderTree/mac/Configurations/Base.xcconfig:
* Tools/DumpRenderTree/mac/Configurations/SDKVariant.xcconfig: Remove.
* Tools/ImageDiff/ImageDiff.xcodeproj/project.pbxproj:
* Tools/ImageDiff/cg/Configurations/Base.xcconfig:
* Tools/ImageDiff/cg/Configurations/SDKVariant.xcconfig: Remove.
* Tools/MiniBrowser/Configurations/Base.xcconfig:
* Tools/MiniBrowser/Configurations/SDKVariant.xcconfig: Remove.
* Tools/MiniBrowser/MiniBrowser.xcodeproj/project.pbxproj:
* Tools/MobileMiniBrowser/Configurations/Base.xcconfig:
* Tools/MobileMiniBrowser/Configurations/SDKVariant.xcconfig: Remove.
* Tools/MobileMiniBrowser/MobileMiniBrowser.xcodeproj/project.pbxproj:
* Tools/Scripts/webkitpy/style/checker.py:
(_all_categories):
(CheckerDispatcher._file_type):
(CheckerDispatcher._create_checker):
* Tools/Scripts/webkitpy/style/checkers/sdkvariant.py: Remove.
* Tools/TestWebKitAPI/Configurations/Base.xcconfig:
* Tools/TestWebKitAPI/Configurations/SDKVariant.xcconfig: Remove.
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/WebEditingTester/Configurations/Base.xcconfig:
* Tools/WebEditingTester/Configurations/SDKVariant.xcconfig: Remove.
* Tools/WebKitTestRunner/Configurations/Base.xcconfig:
* Tools/WebKitTestRunner/Configurations/SDKVariant.xcconfig: Remove.
* Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:
* Tools/lldb/lldbWebKitTester/Configurations/Base.xcconfig:
* Tools/lldb/lldbWebKitTester/Configurations/SDKVariant.xcconfig: Remove.
* WebKit.xcworkspace/contents.xcworkspacedata:

Canonical link: <a href="https://commits.webkit.org/253857@main">https://commits.webkit.org/253857@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f22f5e28bb580329a2f186401f589ad0f7467bb7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/87240 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/31327 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/18062 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/96332 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/149816 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/91215 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/29692 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/25912 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/79350 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/91247 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/92856 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/68/builds/23993 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/74042 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/23855 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/78962 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/79202 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/66869 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/79101 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/27414 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/13038 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/72777 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/27362 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/14053 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/25927 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/2717 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/29046 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/36912 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/75593 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/28984 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/33330 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/16742 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

aproskuryakov reviewed (changes requested):
I think that this is pretty much what we wanted to do, but this patch will break production builds as is. You need to modify the appropriate `Makefile` to copy this new directory in `installsrc`. Please also test that it actually works.

aproskuryakov commented on Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
```diff
@@ -4648,7 +4648,6 @@
 		A1712B4011C7B235007A5315 /* RegExpKey.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = RegExpKey.h; sourceTree = "<group>"; };
 		A18193E11B4E0CDB00FC1029 /* IntlCollatorConstructor.lut.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = IntlCollatorConstructor.lut.h; sourceTree = "<group>"; };
 		A18193E21B4E0CDB00FC1029 /* IntlCollatorPrototype.lut.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = IntlCollatorPrototype.lut.h; sourceTree = "<group>"; };
-		A1B89B88221E002A00EB4CEA /* SDKVariant.xcconfig */ = {isa = PBXFileReference; lastKnownFileType = text.xcconfig; path = SDKVariant.xcconfig; sourceTree = "<group>"; };
```
I just realized that this patch only removes `SDKVariant.xcconfig` from projects, but doesn't add it in the new location. This won't break the build, but probably isn't what we want?

ddkilzer replied:
It adds it to `WebKit.xcworkspace` via the `Configurations` folder, as noted in the commit message:

> - Add Configurations folder to WebKit.xcworkspace.

What, specifically, do you propose instead?

ddkilzer replied:
At one point, I left the `SDKVariant.xcconfig` entries in each Xcode project and turned the local files into symlinks back to the single copy, but that didn't make it obvious that there was a single copy now.

aproskuryakov replied:
I don't know what exactly adding the folder to workspace achieves. Does it make the file visible in Xcode left panel, somewhere close to other xcconfigs? The commit message just states what was done, not why or what it achieves from user point of view.

What are the pros and cons of what you did vs. a pseudo folder?

ddkilzer replied:
> What are the pros and cons of what you did vs. a pseudo folder?

AFAICT, you can't create a pseudo-folder outside of an Xcode project in a workspace.  So all comments below are for the read folder currently added to the project.

Pro:  Additional files show up without having to make any further project changes in the WebKit workspace.

Con:  The "other workspace" (which shall not be named per your request) won't show the files because they're not part of a specific Xcode project for WebKit.  (Specifically, if you type in "SDKVariant" in the "other workspace", the WebKit copy of the file won't show up.)

So based on this, we should probably pick one WebKit project to add these files to.  Maybe WTF since it has so much other shared code/headers?

ddkilzer replied:
> The commit message just states what was done, not why or what it achieves from user point of view.

Well, I had assumed it was obvious that it was added so that people using Xcode to edit files/projects could still find the file.  But this doesn't work for the "other workspace", so it's probably best to add it to one Xcode project—even though the project won't do anything with them—they're just there to expose them in Xcode for viewing/editing.

ddkilzer replied:
> So based on this, we should probably pick one WebKit project to add these files to. Maybe WTF since it has so much other shared code/headers?

To be clear, I think it's better to use a folder in the WebKit workspace (then add a corresponding folder to the "other" workspace) as it's a bit confusing to arbitrarily put references in the WTF project.

Commit:
Unify copies of SDKVariant.xcconfig
https://bugs.webkit.org/show_bug.cgi?id=244374
<rdar://99169843>

Reviewed by Alexey Proskuryakov.

- Create Configurations folder and move a copy of
  SDKVariant.xcconfig into it.
- Add Configurations folder to WebKit.xcworkspace.
- Update `installsrc` target in Makefile to copy Configurations
  folder.
- Remove duplicate copies of SDKVariant.xcconfig in each
  project.
- Remove SDKVariant.xcconfig from Xcode projects.
- Change references to SDKVariant.xcconfig to use a relative
  path to the single copy.
- Remove SDKVariantChecker from check-webkit-style since it's no
  longer needed.

* Configurations/SDKVariant.xcconfig: Rename from Source/JavaScriptCore/Configurations/SDKVariant.xcconfig.
* Makefile:
(installsrc):
* Source/JavaScriptCore/Configurations/Base.xcconfig:
* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/ThirdParty/ANGLE/ANGLE.xcodeproj/project.pbxproj:
* Source/ThirdParty/ANGLE/Configurations/Base.xcconfig:
* Source/ThirdParty/ANGLE/Configurations/SDKVariant.xcconfig: Remove.
* Source/ThirdParty/gtest/xcode/Config/General.xcconfig:
* Source/ThirdParty/gtest/xcode/Config/SDKVariant.xcconfig: Remove.
* Source/ThirdParty/libwebrtc/Configurations/Base.xcconfig:
* Source/ThirdParty/libwebrtc/Configurations/SDKVariant.xcconfig: Remove.
* Source/ThirdParty/libwebrtc/libwebrtc.xcodeproj/project.pbxproj:
* Source/WTF/Configurations/Base.xcconfig:
* Source/WTF/Configurations/SDKVariant.xcconfig: Remove.
* Source/WTF/WTF.xcodeproj/project.pbxproj:
* Source/WebCore/Configurations/Base.xcconfig:
* Source/WebCore/Configurations/SDKVariant.xcconfig: Remove.
* Source/WebCore/PAL/Configurations/Base.xcconfig:
* Source/WebCore/PAL/Configurations/SDKVariant.xcconfig: Remove.
* Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebGPU/Configurations/Base.xcconfig:
* Source/WebGPU/Configurations/SDKVariant.xcconfig: Remove.
* Source/WebGPU/WebGPU.xcodeproj/project.pbxproj:
* Source/WebInspectorUI/Configurations/Base.xcconfig:
* Source/WebInspectorUI/Configurations/SDKVariant.xcconfig: Remove.
* Source/WebInspectorUI/WebInspectorUI.xcodeproj/project.pbxproj:
* Source/WebKit/Configurations/Base.xcconfig:
* Source/WebKit/Configurations/SDKVariant.xcconfig: Remove.
* Source/WebKit/SwiftOverlay/Configurations/Base.xcconfig:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/mac/Configurations/Base.xcconfig:
* Source/WebKitLegacy/mac/Configurations/SDKVariant.xcconfig: Remove.
* Source/bmalloc/Configurations/Base.xcconfig:
* Source/bmalloc/Configurations/SDKVariant.xcconfig: Remove.
* Source/bmalloc/bmalloc.xcodeproj/project.pbxproj:
* Tools/ContentExtensionTester/Configurations/Base.xcconfig:
* Tools/ContentExtensionTester/Configurations/SDKVariant.xcconfig: Remove.
* Tools/DumpRenderTree/DumpRenderTree.xcodeproj/project.pbxproj:
* Tools/DumpRenderTree/mac/Configurations/Base.xcconfig:
* Tools/DumpRenderTree/mac/Configurations/SDKVariant.xcconfig: Remove.
* Tools/ImageDiff/ImageDiff.xcodeproj/project.pbxproj:
* Tools/ImageDiff/cg/Configurations/Base.xcconfig:
* Tools/ImageDiff/cg/Configurations/SDKVariant.xcconfig: Remove.
* Tools/MiniBrowser/Configurations/Base.xcconfig:
* Tools/MiniBrowser/Configurations/SDKVariant.xcconfig: Remove.
* Tools/MiniBrowser/MiniBrowser.xcodeproj/project.pbxproj:
* Tools/MobileMiniBrowser/Configurations/Base.xcconfig:
* Tools/MobileMiniBrowser/Configurations/SDKVariant.xcconfig: Remove.
* Tools/MobileMiniBrowser/MobileMiniBrowser.xcodeproj/project.pbxproj:
* Tools/Scripts/webkitpy/style/checker.py:
(_all_categories):
(CheckerDispatcher._file_type):
(CheckerDispatcher._create_checker):
* Tools/Scripts/webkitpy/style/checkers/sdkvariant.py: Remove.
* Tools/TestWebKitAPI/Configurations/Base.xcconfig:
* Tools/TestWebKitAPI/Configurations/SDKVariant.xcconfig: Remove.
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/WebEditingTester/Configurations/Base.xcconfig:
* Tools/WebEditingTester/Configurations/SDKVariant.xcconfig: Remove.
* Tools/WebKitTestRunner/Configurations/Base.xcconfig:
* Tools/WebKitTestRunner/Configurations/SDKVariant.xcconfig: Remove.
* Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:
* Tools/lldb/lldbWebKitTester/Configurations/Base.xcconfig:
* Tools/lldb/lldbWebKitTester/Configurations/SDKVariant.xcconfig: Remove.
* WebKit.xcworkspace/contents.xcworkspacedata:

Canonical link: https://commits.webkit.org/253857@main
